File: 백준/Bronze/1259. 팰린드롬수/팰린드롬수.java
import java.io.*;
public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		while(true) {
			String str = br.readLine();
			if(str.equals("0")) {
				break;
			}
			
			String reversed = new StringBuilder(str).reverse().toString();
			
			if(str.equals(reversed)) {
				System.out.println("yes");
			} else {
				System.out.println("no");
			}
		}
	}
}

File: 백준/Bronze/1259. 팰린드롬수/팰린드롬수.kt
import java.io.*;
fun main() {
    val br = BufferedReader(InputStreamReader(System.`in`))

    while(true) {
        val str = br.readLine();
        if(str.equals("0")) {
            break;
        }

        val reverseStr = StringBuilder(str).reverse().toString()

        if(str.equals(reverseStr)) {
            println("yes")
        } else {
            println("no")
        }
    }
}